src/features/defi/providers/cosmos/components/CosmosManager/Deposit/components/Confirm.tsx

```diff
@@ -0,0 +1,172 @@
+import { StepComponentProps } from 'components/DeFi/components/Steps'
+import { Row } from 'components/Row/Row'
+import { RawText, Text } from 'components/Text'
+import { useBrowserRouter } from 'hooks/useBrowserRouter/useBrowserRouter'
+import { useWallet } from 'hooks/useWallet/useWallet'
+import { bnOrZero } from 'lib/bignumber/bignumber'
+import {
+  selectAssetById,
+  selectMarketDataById,
+  selectPortfolioCryptoHumanBalanceByAssetId,
+} from 'state/slices/selectors'
+import { useAppSelector } from 'state/store'
+
+import { CosmosDepositActionType } from '../DepositCommon'
+import { DepositContext } from '../DepositContext'
+
+export const Confirm = ({ onNext }: StepComponentProps) => {
+  const { state, dispatch } = useContext(DepositContext)
+  const translate = useTranslate()
+  const { query } = useBrowserRouter<DefiQueryParams, DefiParams>()
+  const { chainId, contractAddress, assetReference } = query
+  const assetNamespace = 'slip44'
+  const assetId = toAssetId({ chainId, assetNamespace, assetReference })
+  const feeAssetId = toAssetId({
+    chainId,
+    assetNamespace: 'slip44',
+    assetReference: ASSET_REFERENCE.Cosmos,
+  })
+
+  const asset = useAppSelector(state => selectAssetById(state, assetId))
+  const feeAsset = useAppSelector(state => selectAssetById(state, feeAssetId))
+  const feeMarketData = useAppSelector(state => selectMarketDataById(state, feeAssetId))
+
+  // user info
+  const { state: walletState } = useWallet()
+
+  // notify
+  const toast = useToast()
+
+  const feeAssetBalance = useAppSelector(state =>
+    selectPortfolioCryptoHumanBalanceByAssetId(state, { assetId: feeAsset?.assetId ?? '' }),
+  )
+
+  const marketData = useAppSelector(state => selectMarketDataById(state, assetId))
+
+  const { handleStakingAction } = useStakingAction()
+
+  if (!state || !dispatch) return null
+
+  const handleDeposit = async () => {
+    if (!state.userAddress || !assetReference || !walletState.wallet) return
+    dispatch({ type: CosmosDepositActionType.SET_LOADING, payload: true })
+
+    const { gasLimit, gasPrice } = await getStakingFees(asset, marketData.price)
+
+    try {
+      const broadcastTxId = await handleStakingAction({
+        asset,
+        validator: contractAddress,
+        chainSpecific: {
+          gas: gasLimit,
+          fee: bnOrZero(gasPrice).times(`1e+${asset?.precision}`).toString(),
+        },
+        value: bnOrZero(state.deposit.cryptoAmount).times(`1e+${asset.precision}`).toString(),
+        action: StakingAction.Stake,
+      })
+
+      dispatch({
+        type: CosmosDepositActionType.SET_DEPOSIT,
+        payload: {
+          txStatus: broadcastTxId?.length ? 'success' : 'failed',
+        },
+      })
+
+      if (!broadcastTxId) {
+        throw new Error() // TODO:
+      }
+
+      dispatch({ type: CosmosDepositActionType.SET_TXID, payload: broadcastTxId })
+    } catch (error) {
+      console.error('CosmosDeposit:handleDeposit error', error)
```

Comment:
moduleLogger

Reply:
Also from the Foxy implementation, will add it in the same stacked PR

Reply:
https://github.com/shapeshift/web/pull/2323